app/components/logo_component.rb

```diff
@@ -0,0 +1,16 @@
+class LogoComponent < ViewComponent::Base
+  def initialize(theme:, classes:)
+    @theme = theme
+    @classes = classes
+  end
+
+  def dark_mode?
+    theme == 'dark'
+  end
+
+  def text_color_class
+    dark_mode? ? 'text-gray-50' : 'text-gray-800'
```

Comment:
Ahh yes, I can see how this is confusing. It's basically used to switch the logo text on both dark and light modes. If we add a text colour to the SVG the logo text will inherit that colour because it has its fill value set to `currentColor`. 

The other option was having two svgs and switching between them in the template, but the only difference would have been the text colour so I went with this approach instead.